dash/include/dash/allocator/CollectiveAllocator.h

```diff
@@ -123,12 +126,15 @@ class CollectiveAllocator
   /**
    * Move-assignment operator.
    */
-  self_t & operator=(const self_t && other) noexcept
+  self_t & operator=(self_t && other) noexcept
   {
     // Take ownership of other instance's allocation vector:
     if (this != &other) {
-      std::swap(_allocated, other._allocated);
```

Comment:
> This is especially problematic, as the deallocator of other will free them later.

That was my point. They will be free'd once `other` is destructed. Is that problematic? Since `other` is an `rvalue`, it's lifetime is limited and the destruction will happen soon anyway (most likely after the call). If it is an `xvalue`, `other` is in a _valid but unspecified_ state afterwards, which is OK with the semantics of move. 

After all, this is just a shortcut and I am fine with keeping it the way it's done in you changes.